pkg/resources/block_storage/resource_volume.go

```diff
@@ -0,0 +1,588 @@
+					},
+				},
+			},
+			"state": schema.StringAttribute{
+				MarkdownDescription: "Volume state.",
+				Computed:            true,
+			},
+		},
+		Blocks: map[string]schema.Block{
+			"timeouts": timeouts.Block(ctx, timeouts.Opts{
+				Read: true,
+			}),
+		},
+	}
+}
+
+// Configure sets up resource dependencies.
+func (r *ResourceVolume) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	r.client = req.ProviderData.(*providerConfig.ExoscaleProviderConfig).ClientV3
+}
+
+// Create resources by receiving Terraform configuration and plan data, performing creation logic, and saving Terraform state data.
+func (r *ResourceVolume) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan ResourceVolumeModel
+
+	// Load Terraform plan plan into the model.
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Set timeout.
+	timeout, diags := plan.Timeouts.Create(ctx, config.DefaultTimeout)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	ctx, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+
+	// Use API endpoint in selected zone.
+	client, err := utils.SwitchClientZone(
+		ctx,
+		r.client,
+		exoscale.ZoneName(plan.Zone.ValueString()),
+	)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"unable to change exoscale client zone",
+			err.Error(),
+		)
+		return
+	}
+
+	// Prepare API request to create a resource.
+	// All required attributes must be present in the request.
+	// Optional attributes not defined in the plan should leave default value in request,
+	// for egoscale to interpret it as unset value and handle it accordingly.
+
+	request := exoscale.CreateBlockStorageVolumeRequest{
+		Name: plan.Name.ValueString(),
+		Size: plan.Size.ValueInt64(),
+	}
+
+	if len(plan.Labels.Elements()) > 0 {
+		labels := exoscale.Labels{}
+
+		dg := plan.Labels.ElementsAs(ctx, &labels, false)
+		if dg.HasError() {
+			resp.Diagnostics.Append(dg...)
+			return
+		}
+
+		request.Labels = labels
+	}
+
+	if !plan.SnapshotTarget.IsNull() {
+		snapshot := VolumeSnapshotTargetModel{}
+
+		dg := plan.SnapshotTarget.As(ctx, &snapshot, basetypes.ObjectAsOptions{})
+		if dg.HasError() {
+			resp.Diagnostics.Append(dg...)
+			return
+		}
+
+		if !snapshot.ID.IsNull() {
+			uuid, err := exoscale.ParseUUID(snapshot.ID.ValueString())
+			if err != nil {
+				resp.Diagnostics.AddError(
+					"unable to parse snapshot ID",
+					err.Error(),
+				)
+				return
+			}
+
+			request.BlockStorageSnapshot = &exoscale.BlockStorageSnapshotTarget{
+				ID: uuid,
+			}
+		}
+	}
+
+	op, err := client.CreateBlockStorageVolume(ctx, request)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to create Volume",
+			err.Error(),
+		)
+		return
+	}
+
+	_, err = client.Wait(ctx, op, exoscale.OperationStateSuccess)
```

Comment:
```suggestion
	op, err = client.Wait(ctx, op, exoscale.OperationStateSuccess)
```

Probably you can update the op variable here to make sure the ID reference is valid and set